orchestra/tests/test_task_lifecycle.py

```diff
@@ -172,117 +177,246 @@ def test_worker_has_reviewer_status(self):
+            }
+
+            assignments = [
+                {
+                    'worker': workers['entry'],
+                    'start_datetime': test_start + timedelta(hours=1)
+                },
+                {
+                    'worker': workers['reviewer'],
+                    'start_datetime': test_start + timedelta(hours=3)
+                }
+            ]
+
+            snapshots = [
+                {
+                    'worker': workers['entry'],
+                    'type': TaskAssignment.SnapshotType.SUBMIT,
+                    'datetime': test_start + timedelta(hours=2),
+                },
+                {
+                    'worker': workers['reviewer'],
+                    'type': TaskAssignment.SnapshotType.REJECT,
+                    'datetime': test_start + timedelta(hours=4),
+                },
+                {
+                    'worker': workers['entry'],
+                    'type': TaskAssignment.SnapshotType.SUBMIT,
+                    'datetime': test_start + timedelta(hours=5),
+                },
+                {
+                    'worker': workers['reviewer'],
+                    'type': TaskAssignment.SnapshotType.ACCEPT,
+                    'datetime': test_start + timedelta(hours=6),
+                },
+            ]
+
+            for i, assignment in enumerate(assignments):
+                TaskAssignment.objects.create(
+                    worker=assignment['worker'], task=task,
+                    status=TaskAssignment.Status.SUBMITTED,
+                    in_progress_task_data={},
+                    assignment_counter=i,
+                    snapshots={'snapshots': [], '__version__': 2},
+                    start_datetime=assignment['start_datetime'])
+
+            for i, snapshot in enumerate(snapshots):
+                assignment = task.assignments.get(worker=snapshot['worker'])
+                assignment.snapshots['snapshots'].append({
+                    'datetime': snapshot['datetime'],
+                    'type': snapshot['type'],
+                    'work_time_seconds': 0,
+                    'data': {
+                        snapshot['worker'].user.username: snapshot['type']
+                    },
+                })
+                assignment.save()
+
+            return task
+
+        def _test_reverted_task(datetime, status, num_assignments,
+                                num_snapshots_per_assignment):
+            task = _sample_complete_task()
+            revert_task_to_datetime(task.id, datetime)
+            task = Task.objects.get(id=task.id)
+            self.assertEquals(task.status, status)
+
+            assignments = assignment_history(task)
+            self.assertEquals(task.assignments.count(), num_assignments)
+            self.assertEquals(
+                len(num_snapshots_per_assignment), num_assignments)
+            for i, num_snapshots in enumerate(num_snapshots_per_assignment):
+                self.assertEquals(
```

Comment:
sanity check: did you also test by hand that this is the case (reverting to a state means the previous worker sees all of their work preserved?